### PR TITLE
New combat layout

### LINKED ISSUE #42: Change combat from list view to a card view

### PR DESCRIPTION
**Pull request by morkusporkus**
Closes #42

### CODE REVIEW

File: src/Maui/MauiProgram.cs [hunk 1]
```diff
@@ -39,6 +39,7 @@ public static MauiApp CreateMauiApp()
         using var scope = app.Services.CreateScope();
 
         var dbContext = scope.ServiceProvider.GetRequiredService<ApplicationDbContext>();
+        dbContext.Database.EnsureDeleted();
```
Comment (benjaminsampica) [inline src/Maui/MauiProgram.cs:42]: Did you mean to keep this

File: src/Maui/Pages/Combat.razor [hunk 2]
```diff
@@ -1,83 +1,73 @@
...
 
     @inject ApplicationDbContext DBContext;
     string returnCreature { get; set; }
     IEnumerable<Creature> creatures;
     List<InitiatedCreature> InitiatedCreatures = new();
+    InitiatedCreature clickedCreature = new();
     Creature selectedCreature = new();
     MudChip selected;
+    bool open;
+    Anchor anchor;
```
Comment (benjaminsampica) [inline src/Maui/Pages/Combat.razor:66]: What is this?

File: src/Maui/Pages/Combat.razor [hunk 3]
```diff
@@ -1,83 +1,73 @@
...
-    @using DMPowerTools.Maui.DataModels;
-    @using DMPowerTools.Maui.Data;
     @using Microsoft.EntityFrameworkCore;
     @using Markdig;
 
     @inject ApplicationDbContext DBContext;
     string returnCreature { get; set; }
     IEnumerable<Creature> creatures;
     List<InitiatedCreature> InitiatedCreatures = new();
+    InitiatedCreature clickedCreature = new();
```
Comment (benjaminsampica) [inline src/Maui/Pages/Combat.razor:62]: I think this is nullable right?

File: src/Maui/Pages/Combat.razor [hunk 4]
```diff
@@ -1,83 +1,73 @@
...
+                <MudCardContent>
+                    <MudText Typo="Typo.caption"><b>Initiative Roll:</b>:@activeCreature.InititiveRoll</MudText>
+                    <MudButton Text="Details" Color="Color.Secondary" OnClick="@(() => OpenDrawer(activeCreature))">Details</MudButton>
+                </MudCardContent>
+            </MudCard>
+        </MudItem>
+    }
+    <MudDrawer Width="100%" Height="80%" @bind-Open="@open" Anchor="Anchor.Bottom" Elevation="1" Variant="@DrawerVariant.Temporary">
+        <MudDrawerHeader>
+            <MudText Typo="Typo.h6">@clickedCreature.Creature.Name</MudText>
```
Comment (benjaminsampica) [inline src/Maui/Pages/Combat.razor:32]: Did you want to use the CreatureDetail component? That way we don’t have to recreate the wheel every time.

The CreatureDetail component currently doesn’t have abilities and actions, tho.

**Review by benjaminsampica — APPROVED** [review]

File: src/Maui/Pages/Combat.razor (hunk 2 above)
Comment (morkusporkus) [reply src/Maui/Pages/Combat.razor:66]: anchor isnt needed now, but the open is the trigger for the drawer

File: src/Maui/Pages/Combat.razor (hunk 4 above)
Comment (morkusporkus) [reply src/Maui/Pages/Combat.razor:32]: Not really, I think each selected creature needs to be different

**Review by benjaminsampica — APPROVED** [review]

**Pull request merged**